### ISSUE TITLE
Current Asset 

### ISSUE THREAD

**Issue by esepro** [Muhammad-1990/iqoption#7]
I'm trying to alert  when price touches bollinger bands 
How can i get the current assest
when i add this script to multiple assets like EURUSD,AUDUSD,USDJPY,GBPUSD how to know which asset produced the alert or error

```

instrument { name = "Test", overlay = true, icon = "indicators:BB" }

period = input (6,"front.period", input.integer,  1)
devs   = input (2, "front.newind.stddev", input.integer, 1)
source = input (1, "front.ind.source", input.string_selection, inputs.titles_overlay)
fn     = input (1, "front.newind.average", input.string_selection, averages.titles)

input_group {
    "front.top line",
    top_color = input { default = "#4BFFB5", type = input.color },
    top_width = input { default = 1, type = input.line_width}
}

input_group {
    "front.middle line",
    middle_color = input { default = "#FF6C58", type = input.color },
    middle_width = input { default = 1, type = input.line_width}
}

input_group {
    "front.bottom line",
    bottom_color = input { default = "#4BFFB5", type = input.color },
    bottom_width = input { default = 1, type = input.line_width}
}

input_group {
    "front.newind.adx.fill",
    fill_color = input { default = rgba(75,255,181,0.08), type = input.color },
    fill_visible = input { default = true, type = input.plot_visibility}
}

local sourceSeries = inputs [source]
local averageFunction = averages [fn]

middle = averageFunction (sourceSeries, period)
scaled_dev = devs * stdev (sourceSeries, period)

top = middle + scaled_dev
bottom = middle - scaled_dev

if fill_visible then
    fill (bottom, top, "", fill_color)
end

plot (top, "front.top line", top_color, top_width)
plot (middle, "front.middle line", middle_color, middle_width)
plot (bottom, "front.bottom line", bottom_color, bottom_width)


buy = close[1]<bottom[1] and high[1]>bottom[1]
sell = high[1] > top[1] and close[1] < top[1]



if buy then
    error("BUY")
end

if sell then
    error("SELL")
end

plot_shape(buy, "long", shape_style.triangleup, shape_size.large, 'green', shape_location.belowbar,0,'BUY', 'green')
plot_shape(sell, "short", shape_style.triangledown, shape_size.large, 'red', shape_location.abovebar,0,'SELL', 'red')


```

**Comment by Muhammad-1990** [comment]
Ok, I have a solution but it needs a bit of work. Also it’s based on angular js… will create a repo for it and link back here…

**Comment by esepro** [comment]
thanks i know a bit of nodejs i'm wondering if the angularjs are you referring to the frontend or terminal coz i never used angularjs

**Comment by Muhammad-1990** [comment]
Ok I will rewrite it in nodejs… it’s not that big. It only gets complex once you start coding in your strategies.

**Comment by esepro** [comment]
that would be great I don't have any strategy for now I'm just testing patterns like engulfing and dark cloud cover and morning star and evening star but I want to trade with margin of safety instead of trading when the candle opens 
for example when bullish engulf happens then I want to take trade only when current candle price is less than the high or open price of red candle and  when bearish engulf happens then I want to take trade only when current candle price is higher than the low or close price of green candle  

<img width="1357" alt="screenshot" src="https://user-images.githubusercontent.com/108810896/179354945-a86167d8-9f29-4aed-87d7-6e10a3d685e0.png">

**Comment by Muhammad-1990** [comment]
Ok cool, that is a good start but I would recommend having at least 3 confirmations before going into that trade. As in only take trades in the direction of the trade and then also look at momentum or volume…. I’m a bit busy trying to finish up the “trade entry checklist” website I’m working on so will only have it ready by the end of the week…

**Comment by esepro** [comment]
yeah I will consider 3 confirmations  before going into the trade. I've looked at the trade entry checklist website 
is it only for trading manually or for auto trading

**Comment by esepro** [comment]
Sound's good looking forward to it

**Comment by usm007** [comment]
how to use your script?? bcz i copied one script to iqoptions but couldnt apply/save it.

**Comment by Muhammad-1990** [comment]
> how to use your script?? bcz i copied one script to iqoptions but couldnt apply/save it.

Hi,
Which script are you having issues with?


all that's needed is to paste the script as a new script and then click save. you can then navigate to the indicators tab and then select scripts and then it will be available to insert onto the chart...

regards,

**Comment by aliadsys** [comment]
Hey 👋
Any new updates on your site or indicators ?

**Comment by enrom1994** [comment]
Hey man 
id love to get in contact with you, ive been working on a project but i need some assistance.
not really a coder

**Comment by Muhammad-1990** [comment]
> Hey man
> id love to get in contact with you, ive been working on a project but i need some assistance.
> not really a coder

Sure, you can open a new issue or start a conversation in the discussion tab. Or if it’s a private project you can email me. M.ahmod@live.com

**Comment by ZERRKERR** [comment]
Hi bro do you have any update about your indicators im from Peru
